app/code/Magento/Store/Model/Store.php

```diff
@@ -666,12 +666,13 @@ public function getBaseUrl($type = UrlInterface::URL_TYPE_LINK, $secure = null)
                     throw new \InvalidArgumentException('Invalid base url type');
             }
 
-            if (false !== strpos($url, self::BASE_URL_PLACEHOLDER)) {
+            if (is_string($url) && false !== strpos($url, self::BASE_URL_PLACEHOLDER)) {
```

Comment:
Hello @anzin 
Please move `false` to the right side

```suggestion
            if (is_string($url) && strpos($url, self::BASE_URL_PLACEHOLDER) !== false) {
```

Thank you in advance